examples/energy-management-app/energy-management-common/include/ChargingTargetsMemMgr.h

```diff
@@ -0,0 +1,107 @@
+ * separately. This class handles that allocation.
+ *
+ * When iterating through the chargingTargetSchedules, an index in this list is kept and the
+ * ChargingTargetsMemMgr::Reset must be called so this object knows which day schedule it is tracking.
+ * This will free any previous memory allocated for the day schedule in this object.
+ *
+ * There are then three usage cases:
+ *
+ * 1. When loading the Target from persistent storage. In this scenario, it is not known upfront
+ *    how many chargingTargets are associated with this day schedule so ChargingTargetsMemMgr::AddChargingTarget()
+ *    needs to be called as each individual chargingTarget is loaded from persistent data.
+ *
+ *    Once the chargingTargets for the day schedule have been loaded, ChargingTargetsMemMgr::AllocAndCopy() is
+ *    called to allocate the memory to store the chargingTargets and the chargingTargets are copied.
+ *
+ * 2. When updating a Target and a day schedule is unaffected, the chargingTargets associated with
+ *    day schedule need copying. The following should be called:
+ *
+ *    ChargingTargetsMemMgr::AllocAndCopy(const DataModel::List<const Structs::ChargingTargetStruct::Type> & chargingTargets)
+ *
+ * 3. When in SetTargets, a new list of chargingTargets needs to be added to a day schedule, the following
+ *    should be called:
+ *
+ *    ChargingTargetsMemMgr::AllocAndCopy(const DataModel::DecodableList<Structs::ChargingTargetStruct::DecodableType> &
+ * chargingTargets)
+ *
+ * Having allocated and copied the chargingTargets accordingly, they can be added to a
+ * DataModel::List<const Structs::ChargingTargetStruct::Type as follows:
+ *
+ * chargingTargetsList = DataModel::List<Structs::ChargingTargetStruct::Type>(ChargingTargetsMemMgr::GetChargingTargets(),
+ * ChargingTargetsMemMgr::GetNumChargingTargets());
+ *
+ * All memory allocated by this object is released When the ChargingTargetsMemMgr destructor is called.
+ *
+ */
+
+#pragma once
+
+#include <app/clusters/energy-evse-server/energy-evse-server.h>
+#include <lib/core/CHIPError.h>
+
+#include <app-common/zap-generated/cluster-objects.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace EnergyEvse {
+
+class ChargingTargetsMemMgr
+{
+public:
+    ChargingTargetsMemMgr();
+    ~ChargingTargetsMemMgr();
+
+    void Reset(uint16_t chargingTargetSchedulesIdx);
+    void AddChargingTarget(EnergyEvse::Structs::ChargingTargetStruct::Type & chargingTarget);
+    void AllocAndCopy();
+    void AllocAndCopy(const DataModel::List<const Structs::ChargingTargetStruct::Type> & chargingTargets);
+    CHIP_ERROR AllocAndCopy(const DataModel::DecodableList<Structs::ChargingTargetStruct::DecodableType> & chargingTargets);
+
+    EnergyEvse::Structs::ChargingTargetStruct::Type * GetChargingTargets() const;
+    uint16_t GetNumChargingTargets() const;
+
+private:
+    EnergyEvse::Structs::ChargingTargetStruct::Type * mpChargingTargets[kEvseTargetsMaxNumberOfDays];
```

Comment:
It may be clearer to rename mpChargingTargets to mpListOfDays (or something better) so it doesn't look too similar to mChargingTargets below, or make the one below mDailyChargingTargets - it's quite hard to follow.

Reply:
Did both. Renamed mpChargingTargets to mpListOfDays and renamed mChargingTargets to mDailyChargingTargets and renamed a couple of methods accordingly.